onnx/defs/math/defs.cc

```diff
@@ -623,10 +623,12 @@ All inputs and outputs must have the same data type.
         "T",
         OpSchema::Variadic);
     schema.Output(0, name, "Output tensor.", "T");
-    schema.TypeConstraint(
-        "T",
-        {"tensor(float16)", "tensor(float)", "tensor(double)"},
-        "Constrain input and output types to float tensors.");
+    if (with_type_constraint) {
+      schema.TypeConstraint(
```

Comment:
Can we make the information to construct type constraint an input of this function? For example,
```
std::function<void(OpSchema&)> ElementwiseMultiOpDocGenerator(
    const char* name,  const vector<string> allowed_types = {"tensor(float16)", "tensor(float)", "tensor(double)"})
```
Using a flag is less explicit than passing the right information directly.

Reply:
I removed the type constraint from that function all together, I think that's clearest :)